### PR TITLE
feat(integrations): add salesforce-experience-cloud provider

### PR DESCRIPTION
**Pull request by gorets**
## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [x] I added tests, otherwise the reason is: 
- [x] I added observability, otherwise the reason is:
- [x] I added analytics, otherwise the reason is:

### CODE REVIEW

**Comment by gorets** [comment]
Also need to add changes to the `packages/shared/lib/clients/provider.client.ts` add `case 'salesforce-experience-cloud':` to two places. I tried it, but your precommit hook changed additional rows.

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -2366,6 +2366,17 @@ salesforce-sandbox:
     proxy:
         base_url: ${connectionConfig.instance_url}
     docs: https://docs.nango.dev/integrations/all/salesforce
+salesforce-experience-cloud:
+    auth_mode: OAUTH2
+    authorization_url: https://${connectionConfig.subdomain}.my.site.com/services/oauth2/authorize
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:2398]: my.site.com? is it a real value or a placeholder?
if placeholder it should probably be `https://${connectionConfig.domain}/services

File: packages/shared/providers.yaml [hunk 2]
```diff
@@ -2366,6 +2366,17 @@ salesforce-sandbox:
...
+    token_url: https://${connectionConfig.subdomain}.my.site.com/services/oauth2/token
+    default_scopes:
+        - offline_access
+    token_response_metadata:
+        - instance_url
+    proxy:
+        base_url: ${connectionConfig.instance_url}
+    docs: https://docs.nango.dev/integrations/all/salesforce
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml]: wrong doc link

**Review by bodinsamuel — COMMENTED** [review]
Thanks for your contribution 🙏🏻 
It's missing a few things, documentation and logo (can be a symlink)
You can follow what has been done previously https://github.com/NangoHQ/nango/pull/2578/files

File: packages/shared/providers.yaml (hunk 1 above)
Comment (gorets) [reply packages/shared/providers.yaml:2398]: yes, it is correct domain for `Salesforce Experience Cloud Sites`

image

https://developer.salesforce.com/docs/atlas.en-us.chatterapi.meta/chatterapi/quickstart_communities.htm

File: packages/shared/providers.yaml (hunk 2 above)
Comment (gorets) [reply packages/shared/providers.yaml]: This is part of the Salesforce product as well as salesforce-sandbox. I don't know what link to put here.

**Comment by bodinsamuel** [comment]
I have added what was missing, merging that. Thanks 🙏🏻

**Comment by gorets** [comment]
> Also need to add changes to the `packages/shared/lib/clients/provider.client.ts` add `case 'salesforce-experience-cloud':` to two places. I tried it, but your precommit hook changed additional rows.

Need remember to make these edits, otherwise the token update logic will not work correctly.

**Comment by bodinsamuel** [comment]
> Need remember to make these edits, otherwise the token update logic will not work correctly.

Ah yes, can you share the diff please? ~Also not sure why the precommit wouldn't work :/~ If the precommit adds more stuff push it, no worries 👍🏻

**Comment by gorets** [comment]
[add_salesforce-experience-cloud_provider.patch](https://github.com/user-attachments/files/16579669/add_salesforce-experience-cloud_provider.patch)

**Review by bodinsamuel — APPROVED** [review]

**Comment by bodinsamuel** [comment]
I'll be deployed during the day or tomorrow 👌🏻

**Pull request merged**